### PR TITLE
Convert controller pod files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controller/podgc/gc_controller.go [hunk 1]
```diff
@@ -179,11 +179,11 @@ func (gcc *PodGCController) gcOrphaned(pods []*v1.Pod, nodes []*v1.Node) {
 		if !deletedNodesNames.Has(pod.Spec.NodeName) {
 			continue
 		}
-		klog.V(2).Infof("Found orphaned Pod %v/%v assigned to the Node %v. Deleting.", pod.Namespace, pod.Name, pod.Spec.NodeName)
+		klog.V(2).InfoS("Found orphaned Pod assigned to the Node. Deleting.", "namespace", pod.Namespace, "pod", pod.Name, "node", pod.Spec.NodeName)
```
Comment (yangjunmyfm192085) [inline pkg/controller/podgc/gc_controller.go:182]: can  klog.KObj works?

File: pkg/controller/podgc/gc_controller.go [hunk 2]
```diff
@@ -179,11 +179,11 @@ func (gcc *PodGCController) gcOrphaned(pods []*v1.Pod, nodes []*v1.Node) {
...
 		if err := gcc.deletePod(pod.Namespace, pod.Name); err != nil {
 			utilruntime.HandleError(err)
 		} else {
-			klog.V(0).Infof("Forced deletion of orphaned Pod %v/%v succeeded", pod.Namespace, pod.Name)
+			klog.V(0).InfoS("Forced deletion of orphaned Pod succeeded", "namespace", pod.Namespace, "pod", pod.Name)
```
Comment (yangjunmyfm192085) [inline pkg/controller/podgc/gc_controller.go:186]: ditto

File: pkg/controller/podgc/gc_controller.go [hunk 3]
```diff
@@ -221,18 +221,18 @@ func (gcc *PodGCController) checkIfNodeExists(name string) (bool, error) {
...
-	klog.V(4).Infof("GC'ing unscheduled pods which are terminating.")
+	klog.V(4).InfoS("GC'ing unscheduled pods which are terminating.")
 
 	for _, pod := range pods {
 		if pod.DeletionTimestamp == nil || len(pod.Spec.NodeName) > 0 {
 			continue
 		}
 
-		klog.V(2).Infof("Found unscheduled terminating Pod %v/%v not assigned to any Node. Deleting.", pod.Namespace, pod.Name)
+		klog.V(2).InfoS("Found unscheduled terminating Pod not assigned to any Node. Deleting.", "namespace", pod.Namespace, "pod", pod.Name)
```
Comment (yangjunmyfm192085) [inline pkg/controller/podgc/gc_controller.go:231]: ditto

File: pkg/controller/podgc/gc_controller.go [hunk 4]
```diff
@@ -221,18 +221,18 @@ func (gcc *PodGCController) checkIfNodeExists(name string) (bool, error) {
...
 		if err := gcc.deletePod(pod.Namespace, pod.Name); err != nil {
 			utilruntime.HandleError(err)
 		} else {
-			klog.V(0).Infof("Forced deletion of unscheduled terminating Pod %v/%v succeeded", pod.Namespace, pod.Name)
+			klog.V(0).InfoS("Forced deletion of unscheduled terminating Pod succeeded", "namespace", pod.Namespace, "pod", pod.Name)
```
Comment (yangjunmyfm192085) [inline pkg/controller/podgc/gc_controller.go:235]: ditto

File: pkg/controller/podgc/gc_controller.go (hunk 1 above)
Comment (scottilee) [reply pkg/controller/podgc/gc_controller.go:182]: Fixed.

File: pkg/controller/podgc/gc_controller.go (hunk 2 above)
Comment (scottilee) [reply pkg/controller/podgc/gc_controller.go:186]: Fixed.

File: pkg/controller/podgc/gc_controller.go (hunk 3 above)
Comment (scottilee) [reply pkg/controller/podgc/gc_controller.go:231]: Fixed.

File: pkg/controller/podgc/gc_controller.go (hunk 4 above)
Comment (scottilee) [reply pkg/controller/podgc/gc_controller.go:235]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085  Addressed your comments.

**Comment by yangjunmyfm192085** [comment]
/ok-to-test
/wg structured-logging
part duplicated with https://github.com/kubernetes/kubernetes/pull/100672

**Comment by scottilee** [comment]
Rebased.

**Comment by scottilee** [comment]
Rebase.

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

**Pull request closed without merging**